### PR TITLE
Respond also with json on node registration

### PR DESCRIPTION
**Pull request by Zauberstuhl**
In case someone (me) wants to automate the process you
cannot really tell if it was successful or not since the
current responds is always a JS redirect to somewhere unknown

### CODE REVIEW

**Comment by Zauberstuhl** [comment]
> fatal: unable to access 'https://git.feneas.org/jaywink/federation.git/': server certificate verification failed. CAfile: /etc/ssl/certs/ca-certificates.crt CRLfile: none
Command "git clone -q https://git.feneas.org/jaywink/federation.git /home/travis/virtualenv/python3.6.3/src/federation" failed with error code 128 in None

oh nooo

File: thefederation/views.py [hunk 1]
```diff
@@ -9,12 +9,14 @@
 
 
 def register_view(request, host):
+    json = True if request.content_type == "application/json" else False
```
Comment (jaywink) [inline thefederation/views.py]: You can also write this a bit more pythonic as `json = request.content_type == "application/json`. But no need to change, just a note :)

File: thefederation/views.py [hunk 2]
```diff
@@ -9,12 +9,14 @@
...
     # TODO rate limit this view per caller ip?
-    if not is_valid_hostname:
-        return redirect("/")
-    if poll_node(host):
-        # Success!
+    if is_valid_hostname and poll_node(host):
+        if json:
+            return JsonResponse({'error': None})
```
Comment (jaywink) [inline thefederation/views.py]: I would maybe prefer `result: success` and then `result: error, reason: foobar` but this is ok too :+1:

**Comment by jaywink** [comment]
> > fatal: unable to access 'https://git.feneas.org/jaywink/federation.git/': server certificate verification failed. CAfile: /etc/ssl/certs/ca-certificates.crt CRLfile: none
> > Command "git clone -q https://git.feneas.org/jaywink/federation.git /home/travis/virtualenv/python3.6.3/src/federation" failed with error code 128 in None
> 
> oh nooo
> 
> on a fresh debian installation with python2 and python3 that works flawless:
> `pip install -e git+https://git.feneas.org/jaywink/federation.git@ddfb82fb8862f26682ac01a82428ea675c2a7fc5#egg=federation==0.18.0.1`
> 
> Also I tested my changes manually on my dev machine as well;
> 
> ```
> # success
> $ curl -L -H "Content-Type: application/json" http://172.18.0.4:5000/register/ggg.social
> {"error": null}
> # fail
> $ curl -L -H "Content-Type: application/json" http://172.18.0.4:5000/register/ggg.socia
> {"error": "Invalid hostname or broken nodeinfo!"}
> ```

Yeah this is something related to travis, maybe some extra certs required for it.

File: thefederation/views.py (hunk 2 above)
Comment (Zauberstuhl) [reply thefederation/views.py]: oh I can change it!

**Pull request merged**